unstructured/ingest/v2/processes/connectors/milvus.py

```diff
@@ -0,0 +1,200 @@
+    def get_client(self) -> "MilvusClient":
+        from pymilvus import MilvusClient
+
+        return MilvusClient(**self.get_connection_kwargs())
+
+
+@dataclass
+class MilvusUploadStagerConfig(UploadStagerConfig):
+    pass
+
+
+@dataclass
+class MilvusUploadStager(UploadStager):
+    upload_stager_config: MilvusUploadStagerConfig = field(
+        default_factory=lambda: MilvusUploadStagerConfig()
+    )
+
+    @staticmethod
+    def parse_date_string(date_string: str) -> float:
+        try:
+            timestamp = float(date_string)
+            return timestamp
+        except ValueError:
+            pass
+        return parser.parse(date_string).timestamp()
+
+    @classmethod
+    def conform_dict(cls, data: dict) -> None:
+        datetime_columns = [
+            "data_source_date_created",
+            "data_source_date_modified",
+            "data_source_date_processed",
+            "last_modified",
+        ]
+
+        json_dumps_fields = ["languages", "data_source_permissions_data"]
+
+        # TODO: milvus sdk doesn't seem to support defaults via the schema yet,
+        #  remove once that gets updated
+        defaults = {"is_continuation": False}
+
+        if metadata := data.pop("metadata", None):
+            data.update(flatten_dict(metadata, keys_to_omit=["data_source_record_locator"]))
+        for datetime_column in datetime_columns:
+            if datetime_column in data:
+                data[datetime_column] = cls.parse_date_string(data[datetime_column])
+        for json_dumps_field in json_dumps_fields:
+            if json_dumps_field in data:
+                data[json_dumps_field] = json.dumps(data[json_dumps_field])
+        for default in defaults:
+            if default not in data:
+                data[default] = defaults[default]
+
+    def run(
+        self,
+        elements_filepath: Path,
+        file_data: FileData,
+        output_dir: Path,
+        output_filename: str,
+        **kwargs: Any,
+    ) -> Path:
+        with open(elements_filepath) as elements_file:
+            elements_contents: list[dict[str, Any]] = json.load(elements_file)
+        for element in elements_contents:
+            self.conform_dict(data=element)
+
+        output_path = Path(output_dir) / Path(f"{output_filename}.json")
+        output_path.parent.mkdir(parents=True, exist_ok=True)
+        with output_path.open("w") as output_file:
+            json.dump(elements_contents, output_file, indent=2)
+        return output_path
+
+
+@dataclass
+class MilvusUploaderConfig(UploaderConfig):
+    collection_name: str
+    num_of_processes: int = 4
+
+
+@dataclass
+class MilvusUploader(Uploader):
+    connection_config: MilvusConnectionConfig
+    upload_config: MilvusUploaderConfig
+    connector_type: str = CONNECTOR_TYPE
+
+    def upload(self, content: UploadContent) -> None:
+        file_extension = content.path.suffix
+        if file_extension == ".json":
```

Comment:
At what point does the Stager create a .csv instead of a .json? Please put some comments in for this.

Reply:
It always creates a json, but the uploader can handle either, in case a user wants to run a custom upload stager to use output csv.